juju/juju
# The firewaller is refactored to push ingress changes to the offering model instead of polling it

## Description of change

TL;DR: the firewaller pushes ingress changes to the offering model instead of that model polling the consuming model. This PR needs a follow up which will have the offering model react to the published changes.

There's also lot of code shuffling which accounts for a lot of the line count. The remotefirewaller facade is removed and the ingress address watcher moved to the firewaller facade. The crossmodelrelations facade also gains a new publish ingress changes method.
The firewaller now only needs to start one watcher to monitor address changes, not two. The new code copied across from the delete remote firewaller facade is unit tested whereas the other firewaller tests are jujuconnsuite, so accommodations are made for that.

## QA steps

Deploy a mediawiki-mysql cross model relation and check logs to see that ingress change is published to mysql model.
